lib/wasix/src/runners/wasi_common.rs

```diff
@@ -257,6 +239,120 @@ fn create_dir_all(fs: &dyn FileSystem, path: &Path) -> Result<(), Error> {
     Ok(())
 }
 
+/// A directory that should be mapped from the host filesystem into a WASI
+/// instance (the "guest").
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub struct MappedDirectory {
+    /// The absolute path for a directory on the host filesystem.
+    pub host: std::path::PathBuf,
+    /// The absolute path specifying where the host directory should be mounted
+    /// inside the guest.
+    pub guest: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct MountedDirectory {
+    pub guest: String,
+    pub fs: Arc<dyn FileSystem + Send + Sync>,
+}
+
+impl From<MappedDirectory> for MountedDirectory {
+    fn from(value: MappedDirectory) -> Self {
+        let MappedDirectory { host, guest } = value;
+        let fs: Arc<dyn FileSystem + Send + Sync> =
+            Arc::new(HostFolderFileSystem::new(crate::default_fs_backing(), host));
+
+        MountedDirectory { guest, fs }
+    }
+}
+
+/// A [`FileSystem`] implementation that is scoped to a specific directory on
+/// the host.
+#[derive(Debug, Clone)]
+struct HostFolderFileSystem<F> {
```

Comment:
Why is this in here?

Shouldn't this be in `virtual-fs`?

As something like `ScopedDirectoryFilesystem` or similar?

Reply:
Moved and renamed :+1: